pkg/clients/helm/client.go

```diff
@@ -107,26 +112,100 @@ func NewClient(log logging.Logger, config *rest.Config, namespace string) (Clien
 	}, nil
 }
 
-func (hc *client) pullAndLoadChart(repo, name, version, username, password string) (*chart.Chart, error) {
+func getVersion(chartFileName string) string {
+	versionIndex := strings.LastIndex(chartFileName, "-")
+	return chartFileName[versionIndex+1 : len(chartFileName)-4]
+}
+
+func getChartFileName(dir string) (string, error) {
+	files, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return "", err
+	}
+	if len(files) != 1 {
+		fileNames := make([]string, 0, len(files))
+		for _, f := range files {
+			fileNames = append(fileNames, f.Name())
+		}
+		return "", errors.Errorf(errUnexpectedDirContentTmpl, strings.Join(fileNames, ","))
+	}
+	return files[0].Name(), nil
+}
+
+// Pulls latest chart version. Returns absolute chartFilePath or error.
+// Side effect: sets version of the latest pulled chart to the spec.
```

Comment:
I would prefer explicit return param instead of implicit `side effect` with reference values.

Reply:
I agree completely. i just tried to keep the interface untouched initially, gladly updated :-)